Library/Homebrew/formula_installer.rb

```diff
@@ -1197,6 +1205,22 @@ def previously_fetched_formula
     end
   end
 
+  sig { void }
+  def fetch_bottle_tab
+    @fetch_bottle_tab ||= begin
+      formula.fetch_bottle_tab
+      @bottle_tab_runtime_dependencies = {}
+      formula.bottle_tab_attributes
+             .fetch("runtime_dependencies")
+             .each_with_object(@bottle_tab_runtime_dependencies) do |runtime_dep, hash|
+        name = runtime_dep["full_name"]
+        hash[name] = runtime_dep
+      end
+      @bottle_tab_runtime_dependencies.freeze
```

Comment:
```suggestion
      @bottle_tab_runtime_dependencies = formula
             .bottle_tab_attributes
             .fetch("runtime_dependencies")
             .index_by { |dep| dep["full_name"] }
             .freeze
```

Nit: It looks like we already pull in `.index_by` so this is another option. I think that the formatting would have to be improved though.